amethyst_renderer/src/debug_drawing.rs

```diff
@@ -0,0 +1,63 @@
+use amethyst_core::cgmath::{Point3, Vector3};
+use amethyst_core::specs::{Component, DenseVecStorage};
+use color::Rgba;
+use vertex::PosColorNorm;
+
+/// Debug lines are stored as a start position, a direction and a color.
+///
+/// Storing a direction instead of an end position may not be intuitive,
+/// but is similar to other 'VertexFormat's.
+pub type DebugLine = PosColorNorm;
+
+/// Resource that stores debug lines to be rendered in DebugLinesPass draw pass
+#[derive(Debug, Default)]
+pub struct DebugLines {
+    /// Lines to be rendered
+    pub lines: Vec<DebugLine>,
+}
+
+impl Component for DebugLines {
+    type Storage = DenseVecStorage<Self>;
+}
+
+impl DebugLines {
+    /// Creates a new screen dimensions object with the given width and height.
+    pub fn new() -> DebugLines {
+        DebugLines {
+            lines: Vec::<DebugLine>::new(),
+        }
+    }
+
+    /// Builder method to pre-allocate a number of line.
+    pub fn with_capacity(mut self, capacity: usize) -> Self {
+        self.lines = Vec::<DebugLine>::with_capacity(capacity);
+        self
+    }
+
+    /// Adds a line to be rendered by giving a position and a direction.
+    pub fn add_as_direction(
```

Comment:
As far as I understood, this needs to be called on every frame for a line to be rendered.
Wouldn't something like `render_as_direction` make it clearer?

Reply:
You understood right. 
True, from a user perspective `add` may sound like there is a `remove` equivalent and something should be removed.

What about `draw_direction`? In my opinion `draw`  is shorter and more intuitive. 

I'm hesitant about `draw_as_direction` because it sounds like `direction` and `line` are going to be rendered in different ways, when the only thing that changes is that one uses a direction and the other an explicit end position for the line.

Reply:
That seems ideal to me.